pig/src/main/java/org/apache/iceberg/pig/IcebergPigInputFormat.java

```diff
@@ -68,6 +68,7 @@ public class IcebergPigInputFormat<T> extends InputFormat<Void, T> {
   private List<InputSplit> splits;
 
   IcebergPigInputFormat(Table table, String signature) {
+    LOG.warn("Iceberg Pig is now deprecated. It will be removed in Iceberg 1.8.");
```

Comment:
rather than logging a warning the class itself should have a `@deprecated` annotation

Reply:
When I discussed with @rdblue, I understood that we want both:
* `@deprecated` for developer
* a log warning for users

Reply:
Let me add `@deprecated` too.

Reply:
Is this required for Hive 2 support as well? I thought only updating https://iceberg.apache.org/multi-engine-support/ is needed.

Reply:
I added `@Deprecated`.

@manuzhang the multi-engine-support update make sense for Hive as it's listed on that page. Pig is not mentioned in this page.